src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -1005,7 +1005,7 @@ public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents,
                 int index = 0;
                 while (index < count)
```

Comment:
consider renaming `count` to `length`? Seems easier to read. `count` is typically a variable amount.